From 24fcdcc128d2b0c8705e9cc1f693725b1b90b1a4 Mon Sep 17 00:00:00 2001
Message-ID: <24fcdcc128d2b0c8705e9cc1f693725b1b90b1a4.1761924396.git.yann.dirson@vates.tech>
From: Vates XCP-ng Developer <vates@vates.tech>
Date: Fri, 31 Oct 2025 15:36:05 +0100
Subject: [XEN PATCH] arm: add debug prints for grant table and p2m entries
To: xen-devel@lists.xenproject.org

Add MFN value to gnttab_mark_dirty() warning message and introduce
a debug printk in p2m_get_entry() to trace MFN, validity, and access
permissions. Also, add some extra prints in unmap_grant_ref() to
verify if some tries to unmap console ring grant.

Signed-off-by: Vates XCP-ng Developer <vates@vates.tech>
---
 xen/arch/arm/include/asm/grant_table.h | 2 +-
 xen/arch/arm/mmu/p2m.c                 | 8 ++++++++
 xen/common/grant_table.c               | 3 +++
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/xen/arch/arm/include/asm/grant_table.h b/xen/arch/arm/include/asm/grant_table.h
index d3c518a926..ac7e3badb0 100644
--- a/xen/arch/arm/include/asm/grant_table.h
+++ b/xen/arch/arm/include/asm/grant_table.h
@@ -20,7 +20,7 @@ static inline void gnttab_clear_flags(struct domain *d,
 static inline void gnttab_mark_dirty(struct domain *d, mfn_t mfn)
 {
 #ifndef NDEBUG
-    printk_once(XENLOG_G_WARNING "gnttab_mark_dirty not implemented yet\n");
+    printk_once(XENLOG_G_WARNING "gnttab_mark_dirty not implemented yet: mfn(%#lx)\n", mfn_x(mfn));
 #endif
 }
 
diff --git a/xen/arch/arm/mmu/p2m.c b/xen/arch/arm/mmu/p2m.c
index 1725cca649..fafbafc7fc 100644
--- a/xen/arch/arm/mmu/p2m.c
+++ b/xen/arch/arm/mmu/p2m.c
@@ -523,6 +523,14 @@ mfn_t p2m_get_entry(struct p2m_domain *p2m, gfn_t gfn,
 
         if ( valid )
             *valid = lpae_is_valid(entry);
+
+        printk("%s: mfn(%#lx), v(%d), r(%d) w(%d) nx(%d)\n",
+               __func__,
+               mfn_x(mfn),
+               lpae_is_valid(entry),
+               entry.p2m.read,
+               entry.p2m.write,
+               entry.p2m.xn);
     }
 
 out_unmap:
diff --git a/xen/common/grant_table.c b/xen/common/grant_table.c
index ebe2d2a3ce..9616d36fd0 100644
--- a/xen/common/grant_table.c
+++ b/xen/common/grant_table.c
@@ -1668,6 +1668,9 @@ unmap_grant_ref(
     common->rd = NULL;
     common->mfn = INVALID_MFN;
 
+    printk("%s: host_addr(%#lx) dev_bus_addr(%#lx), handle(%#x)\n",
+           __func__, op->host_addr, op->dev_bus_addr, op->handle);
+
     unmap_common(common);
     op->status = common->status;
 }
-- 
2.47.3

